yt_dlp/extractor/arte.py

```diff
@@ -5,195 +5,220 @@
-                # original version in different language, with partial subtitles in requested language
-                r'VO(?:(?!{0}).+?)?-ST{0}$'.format(l),
-                # original version in different language, with subtitles for the deaf and hard-of-hearing in requested language
-                r'VO(?:(?!{0}).+?)?-STM{0}$'.format(l),
-                # original version in different language, without subtitles
-                r'VO(?:(?!{0}))?$'.format(l),
-                # original version in different language, with partial subtitles in different language
-                r'VO(?:(?!{0}).+?)?-ST(?!{0}).+?$'.format(l),
-                # original version in different language, with subtitles for the deaf and hard-of-hearing in different language
-                r'VO(?:(?!{0}).+?)?-STM(?!{0}).+?$'.format(l),
-            )
-
-            for pref, p in enumerate(PREFERENCES):
-                if re.match(p, versionCode):
-                    lang_pref = len(PREFERENCES) - pref
-                    break
-            else:
-                lang_pref = -1
-            format_note = '%s, %s' % (f.get('versionCode'), f.get('versionLibelle'))
-
-            media_type = f.get('mediaType')
-            if media_type == 'hls':
-                m3u8_formats = self._extract_m3u8_formats(
-                    format_url, video_id, 'mp4', entry_protocol='m3u8_native',
-                    m3u8_id=format_id, fatal=False)
-                for m3u8_format in m3u8_formats:
-                    m3u8_format.update({
+        subtitles = {}
+
+        for stream in config['data']['attributes']['streams']:
+            # official player contains code like `e.get("versions")[0].eStat.ml5`,
+            # which blindly assumes this structure, so I feel emboldened to do as well
+            stream_version = stream['versions'][0]
+            stream_version_code = stream_version['eStat']['ml5']
+
+            lang_pref = -1
+            m = self.__VERSION_CODE_RE.match(stream_version_code)
+            if m:
+                lc = self.__LANG_MAP.get(lang)
+                lang_pref = sum(
+                    pref << i
+                    for i, pref in enumerate(reversed((
+                        m.group('vlang') == lc,   # we prefer voice in the requested language
+                        not m.group('vaud'),      # and not the audio description version
+                        bool(m.group('vo')),      # but if voice is not in the requested language, at least choose the original voice
+                        m.group('stlang') == lc,  # if subtitles are present, we prefer them in the requested language
+                        not m.group('st'),        # but we prefer no subtitles otherwise
+                        not m.group('stm'),       # and we prefer not the hard-of-hearing subtitles if there are subtitles
+                    )))
+                )
+
+            # XXX: probably not worth warning about if no match
```

Comment:
If `stream_version_code` does not fit `__VERSION_CODE_RE`, the format is assigned a default language preference value of −1. Since this is a mostly cosmetic consequence, I am not sure if it is worth warning about, though I took note of it if I ever were to change my mind.